retry_test.go

```diff
@@ -32,3 +34,29 @@ func TestRetry(t *testing.T) {
 		t.Errorf("invalid number of retries: %d", i)
 	}
 }
+
+func TestRetryWithContext(t *testing.T) {
+	const successOn = 3
```

Comment:
With the current code, shouldn't it only have one success, and never be called again?  Why 3?